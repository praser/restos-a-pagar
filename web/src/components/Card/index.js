import styled from 'styled-components';

export const Card = styled.div`
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid #e3e6f0;
  border-radius: 0.35rem;
  box-shadow: 0 0.15rem 1.75rem 0 rgba(58, 59, 69, 0.15);
  display: flex;
  flex-direction: column;
  margin-bottom: 1.5rem;
`;

export const CardHeader = styled.div`
  border-radius: calc(0.35rem - 1px) calc(0.35rem - 1px) 0 0;
  padding-bottom: 1rem;
  padding-top: 1rem;
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: #f8f9fc;
  border-bottom: 1px solid #e3e6f0;
  color: #4e73df;
  font-weight: 700;
`;

export const CardBody = styled.div`
  flex: 1 1 auto;
  padding: 1.25rem;
`;
