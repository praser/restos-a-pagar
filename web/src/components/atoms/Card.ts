import styled from 'styled-components';

interface IProps {
  width?: string;
}

export const Card = styled.div<IProps>`
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid #e3e6f0;
  border-radius: 0.35rem;
  box-shadow: 0 0.15rem 1.75rem 0 rgba(58, 59, 69, 0.15);
  display: flex;
  flex-direction: column;
  width: 100%;
  margin-left: 0.5rem;
  margin-right: 0.5rem;

  &:first-child {
    margin-left: 0;
  }

  &:last-child {
    margin-right: 0;
  }

  ${({ width }) => width && { width }}
`;

export const CardHeader = styled.div`
  border-radius: calc(0.35rem - 1px) calc(0.35rem - 1px) 0 0;
  padding-bottom: 1rem;
  padding-top: 1rem;
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: #f8f9fc;
  border-bottom: 1px solid #e3e6f0;
  color: #5a5c69;
  font-weight: 700;
`;

export const CardBody = styled.div`
  flex: 1 1 auto;
  padding: 1.25rem;
`;

export const CardSmallText = styled.p`
  font-size: 12px;
  color: darkgray;
`;
