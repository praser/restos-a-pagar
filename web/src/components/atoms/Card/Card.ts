import styled from 'styled-components';

interface IProps {
  width?: string;
}

const Card = styled.div<IProps>`
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid #e3e6f0;
  border-radius: 0.35rem;
  box-shadow: 0 0.15rem 1.75rem 0 rgba(58, 59, 69, 0.15);
  display: flex;
  flex-direction: column;
  width: 100%;
  margin-left: 0.5rem;
  margin-right: 0.5rem;

  &:first-child {
    margin-left: 0;
  }

  &:last-child {
    margin-right: 0;
  }

  ${({ width }) => width && { width }}
`;

export default Card;
