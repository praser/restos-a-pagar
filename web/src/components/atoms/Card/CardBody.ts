import styled from 'styled-components';

const CardBody = styled.div`
  flex: 1 1 auto;
  padding: 1.25rem;
`;

export default CardBody;
