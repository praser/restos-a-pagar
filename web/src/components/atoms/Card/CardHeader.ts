import styled from 'styled-components';

const CardHeader = styled.div`
  border-radius: calc(0.35rem - 1px) calc(0.35rem - 1px) 0 0;
  padding-bottom: 1rem;
  padding-top: 1rem;
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: #f8f9fc;
  border-bottom: 1px solid #e3e6f0;
  color: #5a5c69;
  font-weight: 700;
`;

export default CardHeader;
