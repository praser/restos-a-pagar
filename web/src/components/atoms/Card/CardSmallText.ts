import styled from 'styled-components';

const CardSmallText = styled.p`
  font-size: 12px;
  color: darkgray;
`;

export default CardSmallText;
