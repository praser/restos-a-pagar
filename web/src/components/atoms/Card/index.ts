import Card from './Card';
import CardBody from './CardBody';
import CardHeader from './CardHeader';
import CardSmallText from './CardSmallText';

export { Card, CardBody, CardHeader, CardSmallText };
