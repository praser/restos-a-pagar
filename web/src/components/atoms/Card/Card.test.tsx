import React from 'react';

import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import Card from './Card';

describe('Card atom component', () => {
  it('is expected to render the component correctly', () => {
    const card = renderer.create(<Card />).toJSON();
    expect(card).toMatchSnapshot();
  });

  it('is expected to have the right width if specified', () => {
    const { container } = render(<Card width="80px" />);
    expect(container.firstChild).toHaveStyle('width: 80px');
  });
});
