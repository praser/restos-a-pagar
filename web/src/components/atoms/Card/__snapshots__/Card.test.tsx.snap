// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card atom component is expected to render the component correctly 1`] = `
<div
  className="sc-bdfBwQ gXFlcI"
/>
`;
